src/core/operation/bound_space_op.h

```diff
@@ -44,17 +45,22 @@ inline void ApplyBoundingBox(SimObject* sim_object, double lb, double rb) {
 
 /// Keeps the simulation objects contained within the bounds as defined in
 /// param.h
-class BoundSpace : public Operation {
+class BoundSpace : public OperationImpl {
  public:
-  BoundSpace() : Operation("bdm::BoundSpace") {}
+  BoundSpace() {}
   ~BoundSpace() {}
 
+  BoundSpace* Clone() override { return new BoundSpace(*this); }
+
   void operator()(SimObject* sim_object) override {
     auto* param = Simulation::GetActive()->GetParam();
     if (param->bound_space_) {
       ApplyBoundingBox(sim_object, param->min_bound_, param->max_bound_);
     }
   }
+
+ private:
+  static bool registered_;
```

Comment:
Should we add a `BDM_OP_HEADER(class_name)` macro with the following implementation to reduce the amount of boilerplate code?
```
static bool registered_;
public:
  OperationImpl* Clone() override { return new class_name(*this); }
```

Reply:
I thought about it. The only issue I see is that the cloning operation is not always just a simple return of a new object. If the state of the class is more complex, it will require a custom Clone() operator, in which case, the user should not use the macro. But I guess for most users, the macro will suffice.

So the flow would be like:
```
BDM_OP_HEADER(MyOp);
void MyOp::operator()(SimObject* so) override {
...
}
```
Something like that?